packages/common/src/FindContractVersion.js

```diff
@@ -3,16 +3,36 @@ const assert = require("assert");
 /**
  * Get the version and type of a financial contract deployed using the official UMA contract factories.
  * Note: all inputs and outputs are expressed as fixed-point (scaled by 1e18) BNs.
- * @param {Object} web3 instance
+ * @param {Object} web3 instance. This is passed in to re-use the calling context & network of the entry point.
  * @param {string} contractAddress address of the contract in question
  * @return {Object} contract name & version
  */
 async function findContractVersion(contractAddress, web3) {
   assert(web3, "Web3 object must be provided");
   assert(contractAddress, "Contract address must be provided");
-  const contractCode = await web3.eth.getCode(contractAddress);
+
+  // Note: there is an unknown issue in web3.js that means that the `getCode` syntax does not function correctly in
+  // production. However, ethers has proven to work correctly in production. The code below is a patch to still enable
+  // this module to work while we find a better long term solution for the web3.js issue. If running within unit tests
+  // then the web3.js version is required as it is scope according to the unit test.
+  let contractCode;
+  if (require("minimist")(process.argv.slice())?.network?.includes("mainnet")) {
```

Comment:
nit: will this use web3 on testnets, like kovan? Will that break? Have you thought about using the isPublicNetwork function that we use in migrations?

A (potentially) more general way to switch is what we do in other parts of the codebase to determine whether we're in a truffle/hardhat test env:
```js
if (global.web3) {
  // This is run inside of truffle or hardhat test.
else {
  // This is run literally anywhere else.
}
```

Reply:
ye, that's probably cleaner. The solution I had put here was specifically looking for the case where I knew this _did not_ work, which is mainnet. However, you are right that it is likely that this also does not work on kovan & there is no reason really to not generalize it further. 

Using your `if (global.web3) {` is probs the cleanest. I have updated it to your implementation, PTAL.